client/client.go

```diff
@@ -266,7 +285,11 @@ func NewClientWithOpts(ops ...func(*Client) error) (*Client, error) {
 		// this to avoid breaking existing clients but this should be addressed.
 		c.scheme = "https"
 	}
-
+	if c.hijackDialer == nil {
+		c.hijackDialer = func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
We should put all the defaults in the same place. All the other defaults are at the top of this function before running ops.